src/main/java/data/DatabaseController.java

```diff
@@ -0,0 +1,1032 @@
+			stmt.close();
+
+			return floorList;
+		}
+		catch (SQLException e)
+		{
+			e.printStackTrace();
+			return null;
+		}
+	}
+
+	//make node objects from all entries in the node table
+	//dependent on floorlist already being initialized
+	public static Collection<Node> initializeAllNodes(){
+		try
+		{
+			nodeList.clear(); //sanitize nodelist
+			stmt = connection.createStatement();
+
+			ResultSet results = stmt.executeQuery("SELECT * FROM Node " + "");
+
+			ArrayList<Integer> nodeIDs = new ArrayList<Integer>();
+			while(results.next())
+			{
+				int NodeID = results.getInt(1);
+				nodeIDs.add(NodeID);
+			}
+			for(int i=0;i<nodeIDs.size();i++){
+				nodeList.add(makeNodeByID(nodeIDs.get(i)));
+			}
+			for(int i=0;i<nodeList.size();i++){
+				nodeList.get(i).addNeighbors(getNeighbors(nodeList.get(i).getID()));
+			}
+			results.close();
+			stmt.close();
+
+			return nodeList;
+		}
+		catch (SQLException e)
+		{
+			e.printStackTrace();
+			return null;
+		}
+	}
+
+	//make provider objects from all entries in the provider table
+	//dependent on providerlist already being initialized
+	public static Collection<Provider> initializeAllProviders(){
+		try
+		{
+			providerList.clear(); //sanitize providerList
+			stmt = connection.createStatement();
+
+			ResultSet results = stmt.executeQuery("SELECT * FROM Provider " + "");
+
+			ArrayList<Integer> provIDs = new ArrayList<Integer>();
+			while(results.next())
+			{
+				int NodeID = results.getInt(1);
+				provIDs.add(NodeID);
+			}
+			for(int i=0;i<provIDs.size();i++){
+				providerList.add(makeProviderByID(provIDs.get(i)));
+			}
+			results.close();
+			stmt.close();
+
+			return providerList;
+		}
+		catch (SQLException e)
+		{
+			e.printStackTrace();
+			return null;
+		}
+	}
+
+	/*
+	 * make a single node by NodeID
+	 * @param id NodeID of node object to be created
+	 */
+	public static ConcreteNode makeNodeByID(int id){
+		try
+		{
+			stmt = connection.createStatement();
+
+			ResultSet results = stmt.executeQuery("SELECT * FROM Node " +
+					"WHERE NodeID = " + id + "");
+
+			String NodeName = " ";
+			String NodeType = " ";
+			int XCoord = -1;
+			int YCoord = -1;
+			Floor flr = null;
+
+			if(results.next())
+			{
+				NodeName = results.getString(2);
+				NodeType = results.getString(3);
+				XCoord = results.getInt(4);
+				YCoord = results.getInt(5);
+				flr = getFloorByID(results.getInt(6));
+			} else {
+				return null;
+			}
+			ArrayList<String> data = new ArrayList<>();
+			data.add(NodeName);
+			data.add(NodeType);
+			ConcreteNode node = new ConcreteNode(id, data, XCoord, YCoord, flr); //Return new node using node's information
+			results.close();
+			stmt.close();
+
+			return node;
+		}
+		catch (SQLException e)
+		{
+			//TODO: properly handle exceptions
+			e.printStackTrace();
+			return null;
+		}
+	}
+
+	/*
+	 * retrieve a node from nodeList based on id
+	 * @param id NodeID of desired node
+	 */
+	public static Node getNodeByID(int id){
```

Comment:
As a class on the outside, working with no direct access to the database and no prior nodes loaded (from which I could get an ID), how do I get a NodeID? If there is no outside use for this function, it should be protected or package protected instead of public.